File: backend/app/schemas/dataset.py
from pydantic import BaseModel, Field


class DatasetSummaryResponse(BaseModel):
    file_name: str
    row_count: int
    column_count: int
    columns: list[str]
    sample_rows: list[dict[str, str]] = Field(default_factory=list)


class DatasetLoadResponse(BaseModel):
    message: str
    summary: DatasetSummaryResponse

File: frontend/src/types/dataset.ts
export interface DatasetSummary {
  file_name: string;
  row_count: number;
  column_count: number;
  columns: string[];
  sample_rows: Record<string, string>[];
}

export interface DatasetLoadResponse {
  message: string;
  summary: DatasetSummary;
}
